src/components/ErrorBoundary/index.tsx

```diff
@@ -0,0 +1,60 @@
+import log from 'loglevel';
+import * as React from 'react';
+
+import styles from './styles.module.scss';
+
+export type Props = { _log: typeof log };
+
+type State = {
+  error: Error | null;
+  errorInfo: React.ErrorInfo | null;
+};
+
+export class ErrorBoundaryBase extends React.Component<Props, State> {
+  static defaultProps: Props = { _log: log };
```

Comment:
I thought about using `DefaultProps` as that's really where this belongs, but then I felt weird about doing `export class ErrorBoundaryBase extends React.Component<DefaultProps, State>`. I think the most proper and explicit way to do it would be something like:

```tsx
type PublicProps = {};
type DefaultProps = { _log: typeof log };
export type Props = PublicProps & DefaultProps;
 
export class ErrorBoundaryBase extends React.Component<Props, State> {
  static defaultProps: DefaultProps = { _log: log };

  constructor(props: Props) {
    super(props);
    this.state = { error: null, errorInfo: null };
  }
...
```

but I wasn't sure if it was worth doing all of that. What do you think?

Reply:
I ended up adding a mandatory `children` prop to `PublicProps`, which makes the above more reasonable, so that's what I've implemented.